catalyst-gateway/bin/src/rbac/indexing_context.rs

```diff
@@ -22,6 +22,8 @@ pub struct RbacBlockIndexingContext {
     /// A map contains pending data that will be written in the
     /// `catalyst_id_for_stake_address` table.
     addresses: HashMap<StakeAddress, CatalystId>,
+    /// A set contains removed stake addresses during indexing.
+    removed_addresses: HashSet<StakeAddress>,
```

Comment:
Assume the following situation: some stake address was used in the first transaction then it was removed in the second transaction (so it will be added to the `removed_addresses` set) and then added again by the third one. Expected result: the stake address should be written to the database, but with the current implementation it will be filtered.

I think that I failed to explain the situation during our discussion, but I propose to use the previous implementation when the removed addresses where removed from t he `addresses` map.